crypto/modes/gcm128.c

```diff
@@ -27,6 +27,11 @@ typedef size_t size_t_aX;
 # define PUTU32(p,v)     *(u32 *)(p) = BSWAP4(v)
 #endif
 
+/* RISC-V uses C implementation of gmult as a fallback. */
+#if defined(__riscv)
```

Comment:
`__riscv_xlen == 64` should also be checked here IMHO?

Reply:
I do not agree with this change, but the reason is a bit complicated:

The code, with additional context, is:

```
/* RISC-V uses C implementation of gmult as a fallback. */
#if defined(__riscv)
# define INCLUDE_C_GMULT_4BIT
#endif
```

I do not think we should check for `XLEN == 64` here, because a future 32-bit RISC-V implementation of GCM may also want to use the C impl of `gcm_gmult_4bit` as a fallback.

(Other platforms provide a generic (using few/no extensions) fallback ghash or gmult routine in asm, but until someone has a need for a hand-optimized generic rv64i gmult, it is better to use the well-tested C routine as a fallback. Such a routine can always be contributed by someone later on.)

IMO, the correct place to check for `XLEN==64` is before enabling the `rv64i_zbb_zbc` impl specifically.

Reply:
I'm convinced, though I think adding this check for integrity now and removing the check when there is a "future 32-bit RISC-V implementation of GCM" is also acceptable.